include/zim/archive.h

```diff
@@ -729,7 +729,7 @@ namespace zim
    * @param checksToRun The set of checks to perform.
    * @return False if any check fails, true otherwise.
    */
-  bool validate(const std::string& zimPath, IntegrityCheckList checksToRun);
+  bool LIBZIM_API validate(const std::string& zimPath, IntegrityCheckList checksToRun);
```

Comment:
IMHO, it would be better to introduce a separate macro `LIBZIM_PRIVATE_API` or `LIBZIM_TESTING_API` for this purpose.

Reply:
I don't really see what would be the difference. We don't differentiate how we build the library depending if we will unittest it or simply use it.

Either symbols can be used from calling code or not.

Reply:
`LIBZIM_API` suggests that the labelled identifier is part of the `libzim`'s *public* API. That's not the case for some unittested classes and methods. BTW, I just noticed that some of the new appearances of `LIBZIM_API` (like the one on which this comment thread started) are indeed part of the public API, and they should stay such.